### PR TITLE
Retain call ID after invocation completes

### PR DESCRIPTION
**Pull request by mtopolnik**
* Operation.getCallId() on an inactive operation now keeps returning the call ID of the last invocation, until another call ID is set.
* Operation.setCallId() prevents overwriting the call ID on an active operation.
* Operation.deactivate() must now be used to deactivate the operation.
* Operation.hasActiveInvocation() can be used to check activation state of operation.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java [hunk 1]
```diff
@@ -346,7 +346,7 @@ private void notifyWithExceptionWhenTargetIsNull() {
 
     @Override
     public void sendResponse(Operation op, Object response) {
-        if (!RESPONSE_RECEIVED.compareAndSet(this, FALSE, TRUE)) {
+        if (RESPONSE_RECEIVED.getAndSet(this, TRUE)) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java]: why did you change this? The compareAndSet is more clear since it is more obvious you get a one time shot.

Comment (mtopolnik) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/Invocation.java]: I'll revert this.

**Comment by pveentjer** [comment]
:+1:
